utils/get_previous_daily_ci.py

```diff
@@ -57,8 +57,8 @@ def get_last_daily_ci_run(token, workflow_run_id=None, workflow_id=None, commit_
         if commit_sha in [None, ""] and run["status"] == "completed":
             workflow_run = run
             break
-        # if `commit_sha` is specified, and `workflow_run["head_sha"]` matches it, return it.
-        elif commit_sha not in [None, ""] and run["head_sha"] == commit_sha:
+        # if `commit_sha` is specified, return the latest completed run with `workflow_run["head_sha"]` matching the specified sha.
+        elif commit_sha not in [None, ""] and run["head_sha"] == commit_sha and run["status"] == "completed":
```

Comment:
`run["head_sha"] == commit_sha` The commit_sha could be different for AMD and NVIDIA?

Reply:
No, each day, the scheduled CIs (Nvidia / AMD) are triggered at (almost) the same time, and the associated commit SHA is the latest commit SHA on `main` at that time.

For example, see today's run (the caller being triggered)

https://github.com/huggingface/transformers/actions/runs/15432439509
(AMD)

and 

https://github.com/huggingface/transformers/actions/runs/15432421302
(Nvidia)

On those 2 pages, you will see `78d771`